feat: display settings in preferences

yurii-lunha-stansassets opened the pull request:
## Purpose of this PR
Implemented feature #11 

## Testing status
* No tests have been added.

### Manual testing status
Tested manually in using for the Package Manager plugin

## Comments to reviewers
**1.** Added the About page in user preferences (screenshot), which will display "about us" info and at the same time I referenced its code as a simple example of implementing preferences window. What do you think about it?

**2.** Implemented public TabController that can be useful for developing other plugins, and StansAssets.Plugins.Editor.PackageSettingsWindow<> can use it too (instead of ButtonStrip+ScrollView). It would be better to add it in another PR but PackagePreferencesWindow is dependent on it.

About page
![image](https://user-images.githubusercontent.com/117289281/227189367-4a8859c9-7d11-4a91-ad3d-9973d4d1aee2.png)

Reference example
![image](https://user-images.githubusercontent.com/117289281/227190187-fb1b051e-f438-4508-b503-d476ddb72c36.png)

Commit:
feat: the tab controller implemented (public)

Commit:
feat: PackagePreferencesWindow implemented

kostiantyn-koretskyi reviewed (changes requested):
Few changes required

pavlo-klymentenko-legacy reviewed (changes requested):
Good work! But still needs some love.
Added the video + script below with all the details.

https://github.com/StansAssets/com.stansassets.plugins-dev-kit/assets/13795723/bedec466-a737-4f23-96d4-7983efafe01a

0. Switching between Group and page
1. Changes doesnt represent when 2 winows are open
2. TabController mimics base PackageWindow
3. Registration process (attribute)? Activator.CreateInstance?
	3.1 GetPackageInfo should reused for registration (keywords, name etc)
4. PackagePreferencesWindow.cs line 31
5. CONTRIBUTING.md meta spams (foundation package update from **1.0.16** to **1.0.27**)

Commit:
chore: TabController - removed redundant constructor & parameter

Commit:
refactor: restructured and updated registration process of PackagePreferencesWindow

Commit:
chore: pack 'foundation' dependency version updated

yurii-lunha-stansassets commented on com.stansassets.plugins-dev-kit/Editor/UIToolkit/Controls/TabController.cs:
```diff
@@ -0,0 +1,130 @@
+            if (!m_Tabs.Any())
+            {
+                return;
+            }
```
Moved

yurii-lunha-stansassets commented on com.stansassets.plugins-dev-kit/Editor/UIToolkit/PreferencesWindow/AboutPreferencesWindow.cs:
```diff
@@ -0,0 +1,48 @@
+                SettingsScope.User,
+                new[] { "stan", "plugin", "dev", "kit" })
+            {
+                label = "About"
```
Restructured

yurii-lunha-stansassets commented on com.stansassets.plugins-dev-kit/Editor/UIToolkit/PreferencesWindow/PackagePreferencesWindow.cs:
```diff
@@ -0,0 +1,49 @@
+            }
+
+            var packageInfo = GetPackageInfo();
+            rootElement.Q<Label>("display-name").text = packageInfo.displayName.Remove(0, "Stans Assets - ".Length);
```
Restructured

pavlo-klymentenko committed:
chore: removed duplicted code from PackageSettingsWindow and replaced with TabController

Commit:
chore: ListView extension -> refresh/rebuild in compatible mode

Commit:
chore: TabController -> opened method to refresh active tab

Commit:
fix: correctly open the last selected tab for preferences windows

Commit:
chore: b4f0db83 lost meta

Commit:
refactoring: clean up about page

yurii-lunha-stansassets commented:
Fixed - Switching between Group and page
**Not resolved**- Changes doesn't represent when 2 windows are open
Refactored - TabController mimics base PackageWindow
Refactored- Registration process (attribute)? Activator.CreateInstance?
Refactored- GetPackageInfo should reuse for registration (keywords, name, etc)
Refactored- PackagePreferencesWindow.cs line 31
Updated - CONTRIBUTING.md meta spams (foundation package update from 1.0.16 to 1.0.27)

Please take a look at #12 - I need these changes for other features too

pavlo-klymentenko committed:
chore: reused DisplayName from PackageInfo

pavlo-klymentenko commented:
All the changes merged with newer [PR](https://github.com/StansAssets/com.stansassets.plugins-dev-kit/pull/14), that's why this PR is going to be closed.
All credits to @yurii-lunha-stansassets for updates!
